src/raft.rs

```diff
@@ -1761,7 +1857,10 @@ impl<T: Storage> Raft<T> {
                         e.set_entry_type(EntryType::EntryNormal);
                     }
                 }
-                self.append_entry(&mut m.mut_entries());
+                if !self.append_entry(&mut m.mut_entries()) {
+                    // return ProposalDropped when uncommitted size limit is reached
+                    return Err(Error::ProposalDropped);
+                }
```

Comment:
We may need a debug log here /cc @BusyJay

Reply:
perhaps we could put the debug log inside append_entry?

Reply:
I think put the log  near the `ProposalDropped` error is more clear, in other places where `append_entry` return false, we will just panic so doesn't need a log